src/fraction.ts

```diff
@@ -0,0 +1,281 @@
+  }
+
+  /** Divides by value of another Fraction */
+  divide(param1: Fraction | number = 1, param2: number = 1): this {
+    let otherNumerator: number;
+    let otherDenominator: number;
+
+    if (param1 instanceof Fraction) {
+      otherNumerator = param1.numerator;
+      otherDenominator = param1.denominator;
+    } else {
+      otherNumerator = param1;
+      otherDenominator = param2;
+    }
+
+    return this.set(this.numerator * otherDenominator, this.denominator * otherNumerator);
+  }
+
+  /** Simplifies both sides and checks if they are equal. */
+  equals(compare: Fraction | number): boolean {
+    const a = Fraction.__staticFractionA.copy(compare).simplify();
+    const b = Fraction.__staticFractionB.copy(this).simplify();
+
+    return a.numerator === b.numerator && a.denominator === b.denominator;
+  }
+
+  /** Greater than operator. */
+  greaterThan(compare: Fraction | number): boolean {
+    const a = Fraction.__staticFractionB.copy(this);
+    a.subtract(compare);
+    return a.numerator > 0;
+  }
+
+  /** Greater than or equals operator. */
+  greaterThanEquals(compare: Fraction | number): boolean {
+    const a = Fraction.__staticFractionB.copy(this);
+    a.subtract(compare);
+    return a.numerator >= 0;
+  }
+
+  /** Less than operator. */
+  lessThan(compare: Fraction | number): boolean {
+    return !this.greaterThanEquals(compare);
+  }
+
+  /** Less than or equals operator. */
+  lessThanEquals(compare: Fraction | number): boolean {
+    return !this.greaterThan(compare);
+  }
+
+  /** Creates a new copy with this current values. */
+  clone(): Fraction {
+    return new Fraction(this.numerator, this.denominator);
+  }
+
+  /** Copies value of another Fraction into itself. */
+  copy(copy: Fraction | number): this {
+    if (typeof copy === 'number') {
+      return this.set(copy || 0, 1);
+    }
+    return this.set(copy.numerator, copy.denominator);
+  }
+
+  /** Returns the integer component eg. (5/2) == 2 */
+  quotient(): number {
+    return Math.floor(this.numerator / this.denominator);
+  }
+
+  /** Returns the fraction component when reduced to a mixed number */
+  fraction(): number {
+    return this.numerator % this.denominator;
+  }
+
+  /** Calculates absolute value of itself */
+  makeAbs(): this {
+    this.denominator = Math.abs(this.denominator);
+    this.numerator = Math.abs(this.numerator);
+    return this;
+  }
+
+  /** Returns a raw string representation */
+  toString(): string {
+    return `${this.numerator}/${this.denominator}`;
+  }
+
+  /** Returns a simplified string respresentation */
+  toSimplifiedString(): string {
+    return Fraction.__staticFractionTmp.copy(this).simplify().toString();
+  }
+
+  /** Returns string representation in mixed form */
+  toMixedString(): string {
+    let s = '';
+    const q = this.quotient();
+    const f = Fraction.__staticFractionTmp.copy(this);
+
+    if (q < 0) {
+      f.makeAbs().fraction();
```

Comment:
@0xfe this is the only location where abs was called. I do not understand why fraction is called, do you? The same applies to line 255.

I do not understand why this is now presented as a new file and not a renamed one.

Reply:
I'm afraid I don't know the ins and outs of the fraction module to know why it calls fraction.  The guts of this file were stable since before I started using Vexflow, so I've never dived deeply into it.

> I do not understand why this is now presented as a new file and not a renamed one.

This happens far too often on javscript -> typescript conversions I've found.  When I remembered in my own work I tried to make the first commit just the rename with nothing else.  Even a few changes after the rename tends to make git have a hard time figuring it out.

Reply:
`fraction()` should be renamed to `remainder()`, but even so, I think the call is unnecessary. The call to `makeAbs` should be enough.

Re: `.js` -> `.ts`, you can use `git mv` to tell it that it's a renamed file. Sometimes it's heuristics for detecting this fail.